tests/ttnn/unit_tests/operations/test_line_all_gather_nightly.py

```diff
@@ -0,0 +1,263 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import pytest
+from loguru import logger
+import tt_lib as ttl
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from models.utility_functions import skip_for_grayskull, get_devices_for_t3000
+from tests.ttnn.unit_tests.operations.test_all_gather import is_unsupported_case
+from ttnn import ShardTensorToMesh
+
+
+def run_line_all_gather(
```

Comment:
This run function should probably just go in the base test_all_gather file and replace the corresponding function there. The callers of this function can be updated to use the t3k_device_mesh. Then this should be imported to here.